src/components/Appointment/Form/index.js

```diff
@@ -1,47 +1,281 @@
-import React from 'react';
-import { Col, Row, Form, Input, DatePicker, TimePicker, Select } from 'antd';
+import React, { useEffect, useState } from 'react';
+import {
+  Col,
+  Row,
+  Form,
+  Input,
+  DatePicker,
+  TimePicker,
+  Select,
+  Button,
+} from 'antd';
+import { useFormik } from 'formik';
+import NumberFormat from 'react-number-format';
+import { useSelector, useDispatch } from 'react-redux';
+
+import { validationSchema } from '../validationForm';
+import {
+  DATE_FORMAT,
+  TIME_FORMAT,
+  INITIAL_FORM_VALUES,
+  FETCH_STATUS,
+} from 'constant';
 
 import style from './AppointmentForm.module.css';
+import { fetchDepartments } from 'redux/reducers/appointmentsSlice';
+
+const renderOptionDepartments = (departments) => {
+  if (!departments) return null;
+
+  return departments.map((item) => (
+    <Select.Option key={item} value={item}>
+      {item}
+    </Select.Option>
+  ));
+};
+
+const AppointmentForm = (props) => {
+  const {
+    onSubmitAction,
+    onCancelAction,
+    formValues,
+    submitTitle,
+    cancelTitle,
+  } = props;
+
+  const initialValues = formValues || INITIAL_FORM_VALUES;
+
+  const {
+    handleSubmit,
+    handleChange,
+    handleBlur,
+    resetForm,
+    setFieldValue,
+    setFieldTouched,
+    values,
+    touched,
+    errors,
+    isValid,
+  } = useFormik({
+    validateOnMount: true,
+    initialValues,
+    validationSchema,
+    onSubmit: (values) => {
+      onSubmitAction && onSubmitAction(values);
+      resetForm();
+    },
+  });
+
+  const dispatch = useDispatch();
+  const departments = useSelector((state) =>
+    state.appointments.departments.filter((item) => item !== 'All')
+  );
+  const [departmentStatus, setDepartmentStatus] = useState(FETCH_STATUS.IDLE);
+
+  useEffect(() => {
+    const getDepartments = async () => {
+      setDepartmentStatus(FETCH_STATUS.LOADING);
+
+      try {
+        await dispatch(fetchDepartments()).unwrap();
+        setDepartmentStatus(FETCH_STATUS.SUCCEEDED);
+      } catch (e) {
+        setDepartmentStatus(FETCH_STATUS.FAILED);
+      }
+    };
+
+    if (departmentStatus !== FETCH_STATUS.LOADING && !departments.length) {
+      getDepartments();
+    }
+  });
 
-const formatTime = 'HH:mm';
-const formatDate = 'MM/DD/YYYY';
+  const handleCancel = () => {
+    resetForm();
+    onCancelAction && onCancelAction();
+  };
 
-const AppointmentForm = () => {
   return (
-    <Form layout="vertical" autoComplete="off" size="large">
+    <Form layout="vertical" autoComplete="off" onFinish={handleSubmit}>
       <Row justify="space-around">
         <Col span={10}>
-          <Form.Item name="First Name *" label="First Name *">
-            <Input placeholder="Enter first name" />
+          <Form.Item
```

Comment:
It would be nice to move the logic of From.Item in the separate component and reuse it for all form items.